x/ibc/02-client/handler.go

```diff
@@ -77,16 +75,17 @@ func HandleMsgSubmitMisbehaviour(ctx sdk.Context, k keeper.Keeper, msg *types.Ms
 		return nil, err
 	}
 
-	if err := k.CheckMisbehaviourAndUpdateState(ctx, misbehaviour); err != nil {
+	clientState, err := k.CheckMisbehaviourAndUpdateState(ctx, misbehaviour)
```

Comment:
It looks like clientState is returned here for the sole purpose of emitting the clientType as an event? I think you should add a string method `ClientType` on misbehaviour as well so you can just return the clienttype from misbehaviour. And keep this method as it was originally

Reply:
agree ^^

Reply:
should I adjust `UpdateClient` and `CreateClient` to do the same? They both return client states right now

Reply:
client state is unused in `UpdateClient` and `CreateClient` return, I'm removing it from those functions as well